ludwig/models/trainer.py

```diff
@@ -302,10 +305,7 @@ def train_step(
         Returns:
             A tuple of the loss and a dictionary of metrics.
         """
-
-        # Change to self.optimizer.zero_grad(set_to_none=True) once all ludwig platform on pytorch 1.7 or greater
-        for param in self.model.parameters():
-            param.grad = None
+        self.optimizer.zero_grad()
```

Comment:
Do we need to set `set_to_none=True` as the comment suggested?

Reply:
Looks like some of the distributed tests fail with the error: `TypeError: zero_grad() got an unexpected keyword argument 'set_to_none'`. Reverting this back.